EngineLayer/Silac/SilacConversions.cs

```diff
@@ -332,267 +269,602 @@ public static SpectraFileInfo GetHeavyFileInfo(SpectraFileInfo originalFile, Sil
+                        foreach (SpectraFileInfo info in spectraFileInfo)
                         {
-                            peaks.RemoveAt(i);
-                            i--;
+                            tempFileToRecycleDictionary[info] = new List<double>();
                         }
-                        else
+
+                        List<FlashLFQ.Peptide> peptides = kvp.Value;
+                        foreach (FlashLFQ.Peptide peptide in peptides) //assumes sorted old->new
+                        {
+                            foreach (SpectraFileInfo info in spectraFileInfo)
+                            {
+                                tempFileToRecycleDictionary[info].Add(peptide.GetIntensity(info));
+                            }
+                        }
+
+                        //aggregate into LL, HL, and HH if they contain 2+ missed cleavages
+                        foreach (SpectraFileInfo info in spectraFileInfo)
                         {
-                            foreach (var id in peak.Identifications)
+                            List<double> values = tempFileToRecycleDictionary[info];
+                            if (values.Count > 3)
                             {
-                                SilacLabel label = GetRelevantLabelFromBaseSequence(id.BaseSequence, silacLabels);
-                                HashSet<FlashLFQ.ProteinGroup> originalGroups = id.proteinGroups;
-                                List<FlashLFQ.ProteinGroup> updatedGroups = new List<FlashLFQ.ProteinGroup>();
-                                foreach (FlashLFQ.ProteinGroup group in originalGroups)
+                                if (values.Count == 6)
                                 {
-                                    string groupName = group.ProteinGroupName;
-                                    if (label == null) //if light
-                                    {
-                                        updatedGroups.Add(group);
-                                    }
-                                    else
-                                    {
-                                        string labelString = "(" + label.OriginalAminoAcid + label.MassDifference;
-                                        int labelIndex = groupName.IndexOf(labelString);
-                                        if (labelIndex != -1) //labelIndex == 1 if a) 2+ peptides are required per protein or b) somebody broke parsimony
-                                        {
-                                            groupName = groupName.Substring(0, labelIndex);
-                                            updatedGroups.Add(new FlashLFQ.ProteinGroup(groupName, group.GeneName, group.Organism));
-                                        }
-                                    }
+                                    //LLLLL, LLLLH, LLLHH, LLHHH, LHHHH, HHHHHH have n combinations of 1, 5, 10, 10, 5, 1
```

Comment:
I wonder if there is a need for a restriction of protease to the amino acid label. I guess that most people will use trypsin and label on a residue that gets cleaved. But I imagine, that I could not do that in the program. one could do a lys C or arg C that might have multiple labels but not missed cleavages.

Reply:
this could just be a count of the possible label sites instead of missed cleavages

Reply:
The number of label sites is what is used. The missedCleavages nomenclature was meant to aid understanding (because that's what it usually mirrors), but perhaps it simply confuses. I'll change it.